test/acceptance/log.provider.test.ts

```diff
@@ -0,0 +1,90 @@
+// Copyright IBM Corp. 2013,2017. All Rights Reserved.
+// Node module: loopback-next-extension-starter
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import { Client, createClientForApp, sinon } from '@loopback/testlab';
+import { inject } from '@loopback/context';
+import {
+  Application,
+  get,
+  param,
+  SequenceHandler,
+  FindRoute,
+  ParseParams,
+  InvokeMethod,
+  Send,
+  Reject,
+  CoreBindings,
+  ParsedRequest,
+  ServerResponse,
+} from '@loopback/core';
+import { LogComponent, ExtensionStarterBindings, LogFn } from '../../src';
+
+const coreSequenceActions = CoreBindings.SequenceActions;
+
+describe('Logging', () => {
+  let app: Application;
+
+  beforeEach(createApplication);
+  beforeEach(createController);
+  beforeEach(createSequence);
+
+  it('logs request information', async () => {
+    const spy = sinon.spy(console, 'log');
+    const client: Client = createClientForApp(app);
+    await client.get('/?name=John').expect(200, 'Hello John');
+    const expectedLog = 'ms: /?name=John : (John) => Hello John';
+    sinon.assert.calledWith(spy, sinon.match(expectedLog));
+    spy.restore();
```

Comment:
When the assertion few lines above fails, an error is thrown and the rest of the function is not executed. As a result, `spy.restore()` is never called.

Solution: move `sinon.spy` to `beforeEach` and `spy.restore` to `afterEach` hook.